### PR TITLE
Handle JWT token on Server util & Improve sign flow

### PR DESCRIPTION
**Pull request by heartjigi**
Handle JWT token on Server util
 1. When token is expired, update and request again.
 2. If token is invalid or any other server error code 401 without expired, go to login page.

Improve sign flow
 1. On Sign in, if JWT token is valid, go to main page.
  1-1 If someone didn't fill "Personality Info" go to general info.
  1-2 In this case, if someone fill other page, "Career Info, Expertise Info", fill info on each page.
 2. Update my data on complete component ( After finish fill all info )
 3. On edit profile, each page also fill info.

@ethkim @nlpenguin @MechanicKim @SarahhhC @ace0625 @lyw0149 @ochanje210 @ifndefdeadmau5

### CODE REVIEW

File: src/components/Login/Login.js [hunk 1]
```diff
@@ -184,10 +186,12 @@ class Login extends Component {
...
     } else if (profile) {
-      if (profile.status === 200) {
-        Actions.main({ me: profile });
-      } else if (profile.status === 201) {
+      if (profile.status === 201) {
+        Actions.generalInfo({ me: profile });
+      } else if (profile.personality.length === 0) {
         Actions.generalInfo({ me: profile });
+      } else if (profile.status === 200) {
+        Actions.main({ me: profile });
```
Comment (SarahhhC) [inline src/components/Login/Login.js:178]: Why same code in line 49-54 and 189-194???

File: src/components/SignUp/CareerInfo.js [hunk 2]
```diff
@@ -165,11 +169,31 @@ class CareerInfo extends Component {
...
+      this.state.selected[3] = result[0].education_background;
+
+      this.state.checked[0] = true;
+      this.state.checked[1] = true;
+      this.state.checked[2] = true;
+      this.state.checked[3] = true;
+
+      this.forceUpdate();
+    }
+  }
```
Comment (SarahhhC) [inline src/components/SignUp/CareerInfo.js:207]: where's else case?? Never happen?

File: src/components/SignUp/Completed.js [hunk 3]
```diff
@@ -20,7 +21,15 @@ class Completed extends Component {
...
-    Actions.main({ me: this.props.me });
+    UserUtil.getMyProfile(this.onGetProfileCallback.bind(this));
+  }
+
+  onGetProfileCallback(profile, error) {
+    if (error) {
+      alert('Sever error(Profile)! Please sign in again.');
+    } else if (profile) {
+      Actions.main({ me: profile });
+    }
```
Comment (SarahhhC) [inline src/components/SignUp/Completed.js:32]: Consider else case!

File: src/utils/ApiMeta.js [hunk 4]
```diff
@@ -21,6 +21,11 @@ export const UrlMeta = {
   API_MENTOR_RESP: 'match/response/',
   API_GET_SURVEY_MENTEE: 'survey/request/mentee',
   API_GET_SURVEY_MENTOR: 'survey/request/mentor',
+  API_GET_CAREER: 'survey/request/mentor',
```
Comment (SarahhhC) [inline src/utils/ApiMeta.js]: Same as line 26, both are `API_GET_CAREER`.

File: src/utils/ApiMeta.js [hunk 5]
```diff
@@ -21,6 +21,11 @@ export const UrlMeta = {
...
+  API_GET_GENERAL: 'users/General/',
```
Comment (SarahhhC) [inline src/utils/ApiMeta.js]: There's no get general api. Where did you use this?

File: src/utils/ApiUtil.js [hunk 6]
```diff
@@ -68,23 +70,76 @@ class ApiUtil {
...
 
+  requestUpdateToken() {
+    let reqSet = {
+      method: 'PUT',
+      headers: {
+        access_token: this.reqSet.headers.access_token,
+      },
+    };
+    fetch(UrlMeta.HOST + UrlMeta.API_TOKEN, reqSet)
+    .then((response)=> {
```
Comment (kayla-kwon) [inline src/utils/ApiUtil.js]:
```
.then((response) => {
```

File: src/utils/LoginMeta.js [hunk 7]
```diff
@@ -1,10 +0,0 @@
-class LoginMeta {
```
Comment (SarahhhC) [inline src/utils/LoginMeta.js]: Why deleted??

File: src/utils/ApiMeta.js [hunk 8]
```diff
@@ -39,6 +44,7 @@ export const UrlMeta = {
 export const ErrorMeta = {
   ERR_NONE: 'Undefined Error',
```
Comment (kayla-kwon) [inline src/utils/ApiMeta.js:43]: How about ordering in alphabetically?

File: src/components/AppProps.js [hunk 9]
```diff
@@ -1,10 +1,10 @@
 import {
+  Alert,
```
Comment (MechanicKim) [inline src/components/AppProps.js:2]: Thank you for fixing it

File: src/components/Login/Login.js (hunk 1 above)
Comment (MechanicKim) [reply src/components/Login/Login.js:178]: @heartjigi you can subtract these lines as one method.

File: src/components/SignUp/CareerInfo.js (hunk 2 above)
Comment (MechanicKim) [reply src/components/SignUp/CareerInfo.js:207]: You can use for statement

for (let i = 0; i < 4; i += 1)
  this.state.checked[i] = true;

Also, I think you can probably find more efficient way.

File: src/components/SignUp/Personality.js [hunk 10]
```diff
@@ -34,17 +35,25 @@ class Personality extends Component {
...
+        Actions.completed({ me: this.props.me });
       }
     }
   }
 
+  onGetPersonalityCallback(result, error) {
+    if (error) {
+      alert(JSON.stringify(error));
+    } else if (result.length !== 0) {
+      console.log(this.state.values);
```
Comment (MechanicKim) [inline src/components/SignUp/Personality.js:48]: Is it necessary?

**Review by MechanicKim — COMMENTED** [review]
I think we should use Alert component because the title of javascript alert is 'alert'. Our application dialog needs a reasonable title for each pages.

File: src/components/userProfile/RequestPage.js [hunk 11]
```diff
@@ -79,41 +79,43 @@ class RequestPage extends Component {
...
+            <View style={styles.dropdownContent}>
+              <Menu style={styles.dropdown}
+                onSelect={(value) => this.setState({ selection: value })}>
+                <MenuTrigger>
+                  <Text>{this.state.selection}</Text>
+                </MenuTrigger>
+                <MenuOptions optionsContainerStyle={styles.dropdownOptions}
+                  renderOptionsContainer={(options) =>
+                    <ScrollView>{options}</ScrollView>}>
+                  <MenuOption value='Study abroad'>
```
Comment (kayla-kwon) [inline src/components/userProfile/RequestPage.js:93]: How about making a resource file for object that contains the name of menus?
(like 'Study abroad', 'Get a new job', ... )
That would be better for maintaining.

File: src/utils/ApiMeta.js [hunk 12]
```diff
@@ -21,6 +21,11 @@ export const UrlMeta = {
   API_MENTOR_RESP: 'match/response/',
   API_GET_SURVEY_MENTEE: 'survey/request/mentee',
   API_GET_SURVEY_MENTOR: 'survey/request/mentor',
+  API_GET_CAREER: 'survey/request/mentor',
+  API_GET_GENERAL: 'users/General/',
+  API_GET_CAREER: 'users/Career/',
```
Comment (kayla-kwon) [inline src/utils/ApiMeta.js]: Why line 26~28 apis use uppercase?

File: src/utils/dropdown/select.js [hunk 13]
```diff
@@ -101,12 +101,13 @@ class Select extends Component {
       styleOption,
       styleText,
       activate,
+      value,
     } = this.props;
     const dimensions = { width, height };
 
     let bgColor = (activate) ? '#ffffff' : '#fafafa';
```
Comment (kayla-kwon) [inline src/utils/dropdown/select.js:108]: Making our own color sheet file would be good.

File: src/components/Login/Login.js (hunk 1 above)
Comment (heartjigi) [reply src/components/Login/Login.js:178]: make two callbacks to one

Comment (heartjigi) [reply src/components/Login/Login.js:178]: And add else condition

File: src/components/SignUp/CareerInfo.js (hunk 2 above)
Comment (heartjigi) [reply src/components/SignUp/CareerInfo.js:207]: for (let i = 0; i < 4; i += 1)
this.state.checked[i] = true;
Actually no work on else condition. just pass..

File: src/components/SignUp/Completed.js (hunk 3 above)
Comment (heartjigi) [reply src/components/SignUp/Completed.js:32]: Also It just pass.. else condition

File: src/components/SignUp/Personality.js (hunk 10 above)
Comment (heartjigi) [reply src/components/SignUp/Personality.js:48]: This will handle by @ifndefdeadmau5

File: src/components/userProfile/RequestPage.js (hunk 11 above)
Comment (heartjigi) [reply src/components/userProfile/RequestPage.js:93]: You mean to make META? 
That sounds cool...
But actually this part in charge of @ace0625 
I just fix dropdown bug.
I think he change that!
Thanks

File: src/utils/ApiMeta.js (hunk 4 above)
Comment (heartjigi) [reply src/utils/ApiMeta.js]: Thanks I changed

File: src/utils/ApiMeta.js (hunk 5 above)
Comment (heartjigi) [reply src/utils/ApiMeta.js]: It cause by "일괄처리" 
I'll delete it

File: src/utils/ApiMeta.js (hunk 12 above)
Comment (heartjigi) [reply src/utils/ApiMeta.js]: From copy & paste, Ill change

File: src/utils/ApiUtil.js (hunk 6 above)
Comment (heartjigi) [reply src/utils/ApiUtil.js]: Thanks! solved!

File: src/utils/dropdown/select.js (hunk 13 above)
Comment (heartjigi) [reply src/utils/dropdown/select.js:108]: Because this is module, I just use Hardcode. 
But I agree with U

Comment (kayla-kwon) [reply src/utils/dropdown/select.js:108]: So.. will you make it?

File: src/utils/ApiMeta.js (hunk 8 above)
Comment (heartjigi) [reply src/utils/ApiMeta.js:43]: solved

File: src/utils/LoginMeta.js (hunk 7 above)
Comment (heartjigi) [reply src/utils/LoginMeta.js]: It merged at ApiMeta

File: src/utils/dropdown/select.js (hunk 13 above)
Comment (heartjigi) [reply src/utils/dropdown/select.js:108]: Maybe... next time!

**Comment by SarahhhC** [comment]
Thanks @heartjigi LGTM 👍

**Comment by ochanje210** [comment]
congrats

**Pull request merged**